### PR TITLE
Add http interceptor

### PR DESCRIPTION
**Pull request by campos20**

### CODE REVIEW

**Comment by campos20** [comment]
This is a WIP since backend should be updated as weel

**Comment by suushiemaniac** [comment]
Nitpick: It seems as if all the individual `if (response.ok)` checks in `EventPickerTable` can be dropped now because the global interceptor guarantees that the response is ok anyways (otherwise an error would be thrown and execution doesn't reach that part of the code)

Not sure if this is considered good practise though

File: tnoodle-ui/src/App.js [hunk 1]
```diff
@@ -2,12 +2,62 @@ import React, { Component } from "react";
...
+    handleHttpError = (error) => {
+        let message = error.statusText || JSON.stringify(error);
+        this.updateErrorMessage(message);
+    };
+
+    updateErrorMessage = (errorMessage) => {
+        // Clear the message after some seconds
+        setTimeout(() => {
+            this.setState({ ...this.state, errorMessage: "" });
+        }, 1000 * 10);
```
Comment (suushiemaniac) [inline tnoodle-ui/src/App.js:43]: Magic number calculation. Better create a constant "errorTimoutInSeconds" somewhere and reference that here

File: tnoodle-ui/src/components/EventPickerTable.jsx [hunk 2]
```diff
@@ -62,23 +58,12 @@ const EventPickerTable = connect(
...
         };
 
-        setFlashMessage = (text, bg, response) => {
-            this.props.updateFlashMessage(text + " " + response.statusText, bg);
-        };
-
         getFormats = () => {
             fetchFormats()
                 .then((response) => {
                     if (response.ok) {
```
Comment (suushiemaniac) [inline tnoodle-ui/src/components/EventPickerTable.jsx:64]: Perhaps the check is not necessary anymore, cf. my global comment

File: tnoodle-ui/src/components/EventPickerTable.jsx [hunk 3]
```diff
@@ -91,13 +76,6 @@ const EventPickerTable = connect(
...
```
Comment (suushiemaniac) [inline tnoodle-ui/src/components/EventPickerTable.jsx:76]: see above (?)

File: tnoodle-ui/src/components/EventPickerTable.jsx [hunk 4]
```diff
@@ -110,13 +88,6 @@ const EventPickerTable = connect(
...
```
Comment (suushiemaniac) [inline tnoodle-ui/src/components/EventPickerTable.jsx:88]: see above (?)

File: tnoodle-ui/src/components/EventPickerTable.jsx [hunk 5]
```diff
@@ -151,11 +122,6 @@ const EventPickerTable = connect(
...
```
Comment (suushiemaniac) [inline tnoodle-ui/src/components/EventPickerTable.jsx:122]: see above (?)

File: tnoodle-ui/src/App.js (hunk 1 above)
Comment (campos20) [reply tnoodle-ui/src/App.js]: Updated, thanks! I just changed the "timeout" part because I think this would make it confusing somehow.

**Comment by campos20** [comment]
> Nitpick: It seems as if all the individual `if (response.ok)` checks in `EventPickerTable` can be dropped now because the global interceptor guarantees that the response is ok anyways (otherwise an error would be thrown and execution doesn't reach that part of the code)
> 
> Not sure if this is considered good practise though

Well, that could cause us to try to download as zip an http message. The interceptor does not change the response or prevent it from going its regular path.

**Comment by suushiemaniac** [comment]
Changes looking good! This new version resets the page entirely when clicking "Download scrambles". Could you change this back so that the state of the page remains?

**Comment by campos20** [comment]
> Changes looking good! This new version resets the page entirely when clicking "Download scrambles". Could you change this back so that the state of the page remains?

Done

**Pull request merged**